typescript/src/entity.ts

```diff
@@ -88,15 +150,47 @@ export class Entity {
     });
   }
 
-  public fromJson(json: any) {
+  private getEntityClass = (name: string) => {
+    return (this.merchi as any)[name];
+  }
+
+  protected checkSameSession = (other?: Entity) => {
+    /* istanbul ignore next */
+    if (other !== undefined && other.merchi !== this.merchi) {
+      /* istanbul ignore next */
```

Comment:
the reason that this line has to be excluded from tests is because it doesn't work properly yet.

solving this particular problem is harder than i thought.